OsmAnd/src/net/osmand/plus/views/mapwidgets/widgets/AlarmWidget.java

```diff
@@ -138,29 +143,60 @@ public boolean updateInfo(DrawSettings drawSettings, boolean drawBitmap) {
 							icon.setImageResource(info.locImgId);
 						}
 					}
-					if (!Algorithms.objectEquals(info.text, cachedText) || cachedRegion != info.region) {
+
+					boolean isRegionChanged = cachedRegion != info.region;
```

Comment:
We follow previous code style cause it's shorter version and simpler.

Reply:
@vshcherb Kindly note, that it is not only code style related.

Most likely the initial implementation has an issue with if statement condition in L#193 (previous L#156):

`if (!Algorithms.objectEquals(info.bottomText, cachedBottomText) || cachedRegion != info.region)`

because the cachedRegion could be already updated in L#144 (previous numbering) above, so the second if statement will not detect the region change.

Reply:
The codestyle and technique we use is very simple. 
1 - if statement that checks all cached variable
2- in case **any** state is changed the whole widget / icon is rerendered / recalculated, so pseudocode is always like

```
if( new_state_A != cached_state_A || new_state_B != cached_state_B || ... ) {
      cached_state_A = new_state_A;
      cached_state_B = new_state_B;
....
<CODE using new_state_A, new_state_B, ... >

}
```